vulnerabilities/importers/apache_httpd.py

```diff
@@ -49,47 +62,110 @@ def updated_advisories(self):
         # skips processing the response further to avoid duplicate work
 
         if create_etag(data_src=self, url=self.url, etag_key="ETag"):
-            data = fetch_xml(self.url)
-            advisories = to_advisories(data)
+            links = fetch_links(self.url)
+            self.set_api()
+            advisories = []
+            for link in links:
+                data = requests.get(link).json()
+                advisories.append(self.to_advisory(data))
             return self.batch_advisories(advisories)
 
         return []
 
+    def to_advisory(self, data):
+        cve = data["CVE_data_meta"]["ID"]
+        descriptions = data.get("description", {}).get("description_data", [])
+        description = None
+        for desc in descriptions:
+            if desc.get("lang") == "eng":
+                description = desc.get("value")
+                break
+
+        severities = []
+        impacts = data.get("impact", [])
+        impact = None
+        for imp in impacts:
+            value = imp.get("other")
+            if value is not None:
+                impact = value
+                break
+        if impact is not None:
+            severities.append(
+                VulnerabilitySeverity(
+                    system=scoring_systems["apache_httpd"],
+                    value=impact,
+                )
+            )
+        reference = Reference(
+            reference_id=cve,
+            url=self.ref_url.format(cve),
```

Comment:
Continuation of above `base_url` suggestion.

ref_url can be eliminated. Instead here do something like
`url= urllib.parse.urljoin(self.base_url, cve, ".json")`

Reply:
> `url= urllib.parse.urljoin(self.base_url, cve, ".json")`

this results to only `https://httpd.apache.org/security/json/CVE-1999-1199`.
Can I use `urllib.parse.urljoin(self.base_url, "{}.json".format(cve))` or 
`urllib.parse.urljoin(self.base_url, f"{cve}.json")` ?

Reply:
The f string one works.